### PR TITLE
[#3] - Add reconnect strategies;

### PR DESCRIPTION
**Pull request by kirilknysh**
Closes https://github.com/21-23/phoenix/issues/3

### CODE REVIEW

File: app/strategies/index.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    fibonacci: require('./fibonacci'),
+    linear: require('./linear'),
+    powerOf2: require('./powerOf2'),
+    random: require('./random'),
+};
+const DEFAULT_STRATEGY = 'const';
+
+module.exports = {
+    createStrategy(name, base) {
+        const make = STRATEGIES[name || DEFAULT_STRATEGY] || STRATEGIES[DEFAULT_STRATEGY];
```
Comment (Shastel) [inline app/strategies/index.js]: `STRATEGIES[name] || STRATEGIES[DEFAULT_STRATEGY];`?

File: app/strategies/linear.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+module.exports = function createLinearStrategy(base) {
+    // base is ignored here
+    let value = base - 1;
```
Comment (Shastel) [inline app/strategies/linear.js]: hmm... base is not ignored ;)

**Review by Shastel — CHANGES_REQUESTED** [review]

**Review by Shastel — APPROVED** [review]

File: app/strategies/index.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+const STRATEGIES = {
+    const: require('./const'),
+    fibonacci: require('./fibonacci'),
+    linear: require('./linear'),
+    powerOf2: require('./powerOf2'),
+    random: require('./random'),
+};
+const DEFAULT_STRATEGY = 'const';
```
Comment (lesnitsky) [inline app/strategies/index.js:8]: why const? AFAIK fibonacci is the best strategy in terms of self-DDOS

Comment (kirilknysh) [reply app/strategies/index.js:8]: Just because it was used before strategies introduction. Could be changed to any existing.

**Pull request merged**